src/components/Organization.vue

```diff
@@ -0,0 +1,469 @@
+			const isReference = this.lastReferenceData === this.flattened.length-1
+
+			const arr = [...this.flattened]
+			const created = this.initializeFields()
+
+			// move hoistedFields value to the new level by swapping
+			this.hoistedFields.forEach((field) => {
+				[ created[field], arr[arr.length-1][field] ] = [ arr[arr.length-1][field], created[field] ]
+			})
+			arr.push(created)
+
+			const obj = this.nest(arr)
+			this.$store.commit('replace', {
+				p: this.value,
+				val: obj,
+			})
+			if (isReference) {
+				this.isReferenceData.push(true)
+			} else {
+				this.isReferenceData.push(false)
+				await this.$nextTick()
+				this.$root.$emit('bv::toggle::collapse', this.domId + '-accordion-' + (this.flattened.length-1))
+			}
+		},
+		remove(level) {
+			const arr = [...this.flattened]
+
+			if (level > 0) {
+				// before removing, merge hoistedFields with the level below
+				this.hoistedFields.forEach((field) => {
+					arr[level-1][field] = Array.from(new Set([ ...(arr[level-1][field] || []), ...(arr[level][field] || []) ]))
+				})
+			}
+
+			arr.splice(level, 1)
+			this.isReferenceData.splice(level, 1)
+			const oldKey = this.keys.splice(level, 1)[0]
+			if (arr.length === 0) { // make sure there is at least one organization in the hierarchy
+				this.$set(this.keys, 0, oldKey+1)
+				arr.push(this.initializeFields())
+			}
+
+			const obj = this.nest(arr)
+			this.$store.commit('replace', {
+				p: this.value,
+				val: obj,
+			})
+		},
+		updateValue() {
+			const obj = this.nest(this.flattened)
+			this.$store.commit('replace', {
+				p: this.value,
+				val: obj,
+			})
+		},
+		getParentForLevel(level) {
+			return level < this.countLevels - 1 ? this.flattened[level + 1] : this.value
+		},
+		getQueryExtraForLevel(level) {
+			const last = this.flattened[level-1]
+			if (level === 0) {
+				return ` AND parent_id:""`
+			} else if (last.identifier && last.identifier.startsWith(referenceIdentifierPrefix)) {
+				const code = last.identifier.slice(referenceIdentifierPrefix.length)
+				return ` AND parent_id:"organization_${code}"`
+			} else {
+				return ""
+			}
+		},
+		getDescriptionForLevel(level) {
+			return this.levels && this.levels[level] ? this.levels[level] : ""
+		},
+		getPlaceholderForLevel(level) {
+			return "Select " + this.getDescriptionForLevel(level)
+		},
+	},
+	computed: {
+		canAddNew() {
+			if (this.lastReferenceData === this.flattened.length - 1) {
+				return this.lastReferenceData < 0 || this.flattened[this.lastReferenceData].identifier
+			} else {
+				return true
+			}
+		},
+		lastReferenceData() {
+			let last = -1
+			for (let i=0; i<this.flattened.length; i++) {
+				if (this.getIsReferenceData(i)) {
+					last = i
+				}
+			}
+			return last
+		},
+		countLevels() {
+			if (!this.value) { return -1 }
+			let recurse = this.value
+			let depth = 0
+			while (this.refField in recurse) {
+				depth++
+				recurse = recurse[this.refField]
+			}
+			return depth + 1
+		},
+		flattened() {
```

Comment:
should we have also a computed value for `flattened.length-1` which would be used to find the index

Reply:
done